Docs/CourseFormat.txt

```diff
@@ -86,13 +86,149 @@ Complete Course Template
 -- Finally, award*, where * is a number, allows you to control how many lives
 -- the player gains from successfully completing a course in Oni mode.
 
+
+
+#SONGSELECT:GROUP=Stepmania 5:BPMRANGE=120..160:Difficulty=Medium:MODS=2x;
+-- #SONGSELECT is a new alternative to the #SONG tag, which will (hopefully)
+-- be easier to use and understand. Instead of relying on values input in a
+-- very specific and poorly-documented order, this tag is composed of 
+-- a series of PARAM=VALUE statements, separated by :'s. 
+
+-- The following parameters are available for selecting a song:
+-- TITLE, GROUP, ARTIST, GENRE, DIFFICULTY, METER, BPMRANGE, DURATION, SORT
+-- With the exception of SORT, these can be used in any combination and order.
+-- (See the SORT parameter description below for a better explanation)
+-- If none are specified, then a song and difficulty will be chosen completely at random.
+-- If a combination is specified that results in no possible songs to select,
+-- then that #SONGSELECT entry is skipped.
+
+-- Parameter Desciptions:
+--  TITLE= A list of one or more song titles, separated by commas.
+--  Notes: 
+--  - The "title" here can be either the name of the song directory,
+--    the #TITLE, or the #TITLETRANSLIT.
+--  - Titles must be an *exact match*.
+--  - Unlike the #SONG tag, this will not take into account group names.
+--    If you want to specify a specific song from a specific group, use
+--    this along with the GROUP parameter.
+--  - This does not support names that have commmas.
```

Comment:
If you add support for escape characters I think you should be able to do this. I suspect you'd also run into issues with titles that have # and ; in it as well.

Reply:
This turned out to be slightly more complicated than I expected, but I just pushed some changes that will handle commas and control characters (`#`, `:`, `;`).